Vwitter/View Controllers/SignupViewController.m

```diff
@@ -0,0 +1,60 @@
+//
+//  SignupViewController.m
+//  Vwitter
+//
+//  Created by Christina Li on 7/8/22.
+//
+
+#import "SignupViewController.h"
+#import "VWUser.h"
+
+@interface SignupViewController ()
+@property (weak, nonatomic) IBOutlet UITextField *usernameField;
+@property (weak, nonatomic) IBOutlet UITextField *screennameField;
+@property (weak, nonatomic) IBOutlet UITextField *passwordField;
+
+@end
+
+@implementation SignupViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view.
+}
+
+- (void)registerUser {
+
+    VWUser *newUser = [VWUser user];
+
+    newUser.username = self.usernameField.text;
+    newUser.password = self.passwordField.text;
+    newUser.screenName = self.screennameField.text;
+
+    [newUser signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * error) {
+        if (error != nil) {
+            NSLog(@"Error: %@", error.localizedDescription);
```

Comment:
One very important thing is when things go wrong, you definitely wanna give the user some kind of feedback/error message, you can easily show errors with the iOS SDK's `UIAlertController` - [here's a stackoverflow example](https://stackoverflow.com/questions/42173060/how-to-use-uialertcontroller)

One thing I might recommend is making it even more general for ALL of the view controllers in your app to be able to present an "error" alert easily by just calling something like this 

`[self presentErrorMessageWithTitle:@"something wrong wrong" message:@"detailed explanation of what went wrong"]`

The way you could easily achieve that is by creating a "category" for `UIViewController`, which adds this method. Then, for any view controller in your app, you can just import that header file (usually, categories are named in the format `Class+MyCategoryName.h`, so in this case, you'd probably name it something like `UIViewController+ErrorAlertPresenter.h`

I like this stack overflow example of adding some "testMethod" to UIImage: https://stackoverflow.com/a/24324417